### DISCUSSION TITLE
How to configure telescope-live-grep-args.nvim

### DISCUSSION THREAD

**Discussion by 1-800-jono** [LunarVim/LunarVim#4217] [Q&A] [answered]
Been looking online but couldn't find an answer to this. The closest example comes from the readme in https://github.com/nvim-telescope/telescope-live-grep-args.nvim. But not sure how to apply it to lunarvim's config. Thanks for the help.

```
use {
  "nvim-telescope/telescope.nvim",
  requires = {
    { "nvim-telescope/telescope-live-grep-args.nvim" },
  },
  config = function()
    require("telescope").load_extension("live_grep_args")
  end
}
```

**Comment by LostInTheLogs** [comment]
alright, so here go the questions:
1. have you read https://www.lunarvim.org/docs/configuration/plugins/user-plugins?

**Comment by 1-800-jono** [comment]
Yes I did. From that I came up with:
```
lvim.plugins = {
 { 
    "nvim-telescope/telescope.nvim",
      requires = {
        { "nvim-telescope/telescope-live-grep-args.nvim" },
      },
      config = function()
        require("telescope").load_extension("live_grep_args")
      end
    }
}
```

and also tried:
```
vim.plugins = {
 { 
    "nvim-telescope/telescope-live-grep-args.nvim" ,
      config = function()
        require("telescope").load_extension("live_grep_args")
      end
}
}
```

But didn't work - also don't know what I'm doing :)

**Reply by LostInTheLogs** [reply]
what's your :LvimVersion ?

**Reply by LostInTheLogs** [reply]
your first code works for me
![image](https://github.com/LunarVim/LunarVim/assets/110467150/e9240b67-5ca2-446d-9990-e437fab53cdf)

**Comment by 1-800-jono** [comment]
`release-1.3/neovim-0.9-0aea5a68`

Using `<leader> s t`

**Reply by LostInTheLogs** [reply]
`<leader> s t` isn't using the extension unless you modified it
here's their readme
![image](https://github.com/LunarVim/LunarVim/assets/110467150/c139c4f2-96f1-4e29-b004-db1c1597d47d)
https://github.com/nvim-telescope/telescope-live-grep-args.nvim#setup

**Comment by 1-800-jono** [comment]
How is that setup done in LunarVim? Or maybe it's done elsewhere?

**Reply by LostInTheLogs** [reply]
just a mapping 
`lvim.builtin.which_key.mappings.s.t = { require('telescope').extensions.live_grep_args.live_grep_args, "Live grep args", }
`

**Reply by 1-800-jono** [reply]
Thanks

**Comment by 1-800-jono** [comment, marked as answer]
For clarity here's the correct answer (Both of these code blocks go in `config.lua`):

```
lvim.plugins = {
    { 
        "nvim-telescope/telescope.nvim",
          requires = {
            { "nvim-telescope/telescope-live-grep-args.nvim" },
          },
          config = function()
            require("telescope").load_extension("live_grep_args")
          end
    }
}
```

If you would like to remap `<leader> s t` to use this then: 
```
lvim.builtin.which_key.mappings.s.t = {
  require('telescope').extensions.live_grep_args.live_grep_args, "Live grep args",
}

```

**Comment by LinoWhy** [comment]
Setting the configuation directly in lvim.plugins overrides the telescope config of lvim.builtin.telescope, refers to:
#3406

A warning of "overriding telescope config" may also occur if you run ":checkhealth".

So in my case, i  put the following code after telescope configuration.
`require("telescope").load_extension("live_grep_args")`
